Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandRotateLogic.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace HoloToolkit.Unity.InputModule.Utilities.Interations
+{
+    /// <summary>
+    /// Implements common logic for rotating holograms using a handlebar metaphor. 
+    /// each frame, object_rotation_delta = rotation_delta(current_hands_vector, previous_hands_vector)
+    /// where hands_vector is the vector between two hand/controller positions.
+    /// 
+    /// Usage:
+    /// When a manipulation starts, call Setup.
+    /// Call Update any time to update the move logic and get a new rotation for the object.
+    /// </summary>
+    public class TwoHandRotateLogic
+    {
+        /// <summary>
+        /// Private variables
+        /// </summary>
+        private const float MinHandDistanceForPitchM = 0.1f;
+        private const float RotationMultiplier = 2f;
+        private readonly AxisConstraint m_rotationConstraint;
+        private Vector3 m_previousHandlebarRotation;
+
+        /// <summary>
+        /// The current rotation constraint might be modified based on disambiguation logic, for example
+        /// XOrYBasedOnInitialHandPosition might change the current rotation constraint based on the 
+        /// initial hand positions at the start
+        /// </summary>
+        private AxisConstraint currentRotationConstraint;
+
+        /// <summary>
+        /// Public Accessor for AxisConstraint Property used to constrain Rotation to an axis.
+        /// </summary>
+        public AxisConstraint CurrentRotationConstraint
+        {
+            get
+            {
+                return currentRotationConstraint;
+            }
+            set
+            {
+                currentRotationConstraint = value;
+            }
+        }
+
+        private Vector3 ProjectHandlebarGivenConstraint(AxisConstraint constraint, Vector3 handlebarRotation, Transform manipulationRoot)
```

Comment:
Please also document private methods. It helps with understanding the code and our doc tools will do the right thing wrt only publishing public and protected docs.